src/Uno.UWP/Collections/MaterializableList.cs

```diff
@@ -12,66 +12,86 @@ namespace Uno.Collections
 	/// THIS IS NOT THREAD-SAFE. It is designed to be used on
 	/// the UI thread.
 	/// </remarks>
-	public class MaterializableList<T> : IList<T>
+	public class MaterializableList<T> : IList<T>, IReadOnlyList<T>
 	{
-		private readonly List<T> _listImplementation = new List<T>();
+		private readonly List<T> _innerList;
+
+		public MaterializableList()
+		{
+			_innerList = new List<T>();
+		}
+
+		public MaterializableList(int capacity)
+		{
+			_innerList = new List<T>(capacity);
+		}
+
+		public MaterializableList(List<T> collection)
+		{
+			_innerList = collection;
+		}
+
+		public MaterializableList(IEnumerable<T> collection)
+		{
+			_innerList = new List<T>(collection);
+		}
 
 		public IEnumerator<T> GetEnumerator() => Materialized.GetEnumerator();
 
 		IEnumerator IEnumerable.GetEnumerator() => Materialized.GetEnumerator();
 
 		public void Add(T item)
 		{
-			_listImplementation.Add(item);
+			_innerList.Add(item);
 			_materialized = null;
 		}
 
 		public void Clear()
 		{
-			_listImplementation.Clear();
+			_innerList.Clear();
 			_materialized = null;
 		}
 
-		public bool Contains(T item) => _listImplementation.Contains(item);
+		public bool Contains(T item) => _innerList.Contains(item);
 
-		public void CopyTo(T[] array, int arrayIndex) => _listImplementation.CopyTo(array, arrayIndex);
+		public void CopyTo(T[] array, int arrayIndex) => _innerList.CopyTo(array, arrayIndex);
 
 		public bool Remove(T item)
 		{
 			_materialized = null;
-			return _listImplementation.Remove(item);
+			return _innerList.Remove(item);
 		}
 
-		public int Count => _listImplementation.Count;
+		public int Count => _innerList.Count;
 
 		public bool IsReadOnly => false;
 
-		public int IndexOf(T item) => _listImplementation.IndexOf(item);
+		public int IndexOf(T item) => _innerList.IndexOf(item);
 
 		public void Insert(int index, T item)
 		{
-			_listImplementation.Insert(index, item);
+			_innerList.Insert(index, item);
 			_materialized = null;
 		}
 
 		public void RemoveAt(int index)
 		{
-			_listImplementation.RemoveAt(index);
+			_innerList.RemoveAt(index);
 			_materialized = null;
 		}
 
 		public T this[int index]
 		{
-			get => _listImplementation[index];
+			get => _innerList[index];
 			set
 			{
-				_listImplementation[index] = value;
+				_innerList[index] = value;
 				_materialized = null;
 			}
 		}
 
 		private List<T> _materialized;
 
-		public List<T> Materialized => _materialized ?? (_materialized = _listImplementation.ToList());
+		public List<T> Materialized => _materialized ?? (_materialized = _innerList.ToList());
```

Comment:
It's really not obvious what this does ... I would convert this as a method and add some green to explain the behavior. Wondering also if the returned value should not be exposed as `IReadOnlyList` (or the name of the property/method should underline it)

Reply:
No the result must be a `List<T>` for performance consideration. Using this is an advanced scenario anyway...

I'll add comments.